server.py

```
@@ -472,47 +474,18 @@ def chat(chat_request: ChatRequest, http_request: Request):
-            # Pause-mode tools require streaming (the pause/resume flow needs SSE)
-            if has_pause_tools and not chat_request.stream:
-                raise HTTPException(
-                    status_code=400,
-                    detail="frontend_tools with mode='pause' requires stream=true (the pause/resume flow needs SSE)",
-                )
+        try:
+            request_ai, has_pause_tools = inject_frontend_tools(
+                ai, chat_request.frontend_tools
+            )
+        except FrontendToolCollisionError as e:
+            raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)